packages/react/src/auto/shadcn/inputs/ShadcnAutoRolesInput.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useCallback, useMemo } from "react";
+import type { Control } from "../../../useActionForm.js";
+import { getPropsWithoutRef } from "../../../utils.js";
+import { autoInput } from "../../AutoInput.js";
+import { useRoleInputController } from "../../hooks/useRoleInputController.js";
+import type { AutoRelationshipInputProps } from "../../interfaces/AutoRelationshipInputProps.js";
+import type { ShadcnElements } from "../elements.js";
+import { makeShadcnAutoComboInput } from "./ShadcnAutoComboInput.js";
+import { makeSelectedRecordTags } from "./relationships/SelectedRelatedRecordTags.js";
+
+export interface EnumOption {
+  label: string;
+  value: string;
+}
+
+type AutoRolesInputProps = {
+  field: string; // Field API identifier
+  control?: Control<any>;
+  label?: string;
+  options: EnumOption[];
+};
+
+export const makeShadcnAutoRolesInput = ({
+  Badge,
+  Button,
+  Command,
+  CommandItem,
+  CommandInput,
+  Label,
+  CommandList,
+  CommandEmpty,
+  CommandGroup,
+  Checkbox,
+}: Pick<
+  ShadcnElements,
+  "Badge" | "Button" | "Command" | "CommandItem" | "CommandList" | "CommandEmpty" | "CommandGroup" | "CommandInput" | "Label" | "Checkbox"
+>) => {
+  const { SelectedRecordTags } = makeSelectedRecordTags({ Badge, Button });
+  const ShadcnComboInput = makeShadcnAutoComboInput({
+    Command,
+    CommandInput,
+    Label,
+    CommandItem,
+    CommandList,
+    CommandEmpty,
+    CommandGroup,
+    Checkbox,
+  });
+
+  function ShadcnAutoRolesInput(props: AutoRelationshipInputProps) {
+    const { field } = props;
+    const { options, loading, rolesError, fieldError, selectedRoleKeys, fieldProps, metadata } = useRoleInputController(props);
+
+    const callOnChange = useCallback(
+      (newValues: string[]) => {
+        fieldProps.onChange(newValues);
+      },
+      [fieldProps]
+    );
+
+    const selectedValues = useMemo(
+      () => (selectedRoleKeys ? (selectedRoleKeys.length > 0 ? selectedRoleKeys : [selectedRoleKeys]) : []),
+      [selectedRoleKeys]
+    );
+    const selectedOptions = options.filter((option) => selectedValues.includes(option.value));
+
+    const onSelect = (record: Record<string, any>) => {
+      const existingValues = selectedOptions?.map((option) => option.value) ?? [];
+      const newValues = existingValues.includes(record.value)
+        ? existingValues.filter((value) => value !== record.value)
+        : [...existingValues, record.value];
+      callOnChange(newValues);
+    };
+
+    const roleOptions = React.useMemo(
+      () =>
+        options?.map((option) => ({
+          id: option.value,
+          label: option.label,
+        })) ?? [],
+      [options]
+    );
+
+    const recordOptions = React.useMemo(
```

Comment:
tiny thing but it'd be good to stay consistent between `React.useMemo` and `useMemo` by itself, I kinda prefer the latter. you dont have to go and fix it, just going forward 👍